tools/test.sh

```diff
@@ -36,8 +36,14 @@ cp -ruvf build/img /srv
 
 cd tests
 export PYTEST_ADDOPTS="${PYTEST_ADDOPTS:-} --pdbcls=IPython.terminal.debugger:TerminalPdb"
-pytest "$@"
-ret=$?
+
+{
+    # disable errexit momentarily so we can capture the exit status
+    set +e
+    pytest "$@"
+    ret=$?
+    set -e
+}
```

Comment:
nit: maybe something like this is simpler?

```suggestion
ret=0
pytest "$@" || ret = $?
```

Reply:
That does look better. For now I will go with the current. I Hope I remember it for the next time :D